codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/PaginatorGenerator.kt

```diff
@@ -148,7 +148,7 @@ class PaginatorGenerator : KotlinIntegration {
                     write("")
                     withBlock("while (hasNextPage) {", "}") {
                         withBlock("val req = initialRequest.copy {", "}") {
-                            write("this.$markerLiteral = cursor")
+                            write("this.$markerLiteral = this.$markerLiteral ?: cursor")
```

Comment:
question / correctness: Won't this logic **always** use the `initialRequest.$markerLiteral` if it's not null? How does the cursor get updated to fetch the second page in that case?

I was expecting the fix to be changing L146 to:

```kt
write("var cursor: #F = initialRequest.#L", cursorSymbol, markerLiteral)
```

This will be `null` when the `initialRequest.$markerLiteral` is not set, preserving existing behavior